pkg/apis/wsecurity/v1/v1.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+// A Profile describing the Value
+type ValueProfile interface {
+	// Profile the data provided in args
+	Profile(args ...interface{})
+
+	// Return a multiline string ready for logging at indentation depth
+	String(depth int) string
```

Comment:
The `depth` argument for `String()` is surprising, since a `String` method without a depth argument will cause the object to automatically meet the `fmt.Stringer` interface, which is pretty useful.

Additionally, we tend to prefer structured (JSON format) logging from components via zap.logger, and it feels like multi-line indentation would be less useful given structured logs.

Reply:
Yep, this is probably a not-so-great choice made earlier in the design and I never got to rethink it.

Let's define the problem at hand:
GuardianConfig includes a ReqConfig which includes a list of HeadersConfig, each header has a key and a value, each value has a list of counters and each counter has a list of ranges (MinMax). 

It is very useful to be able to print a GuradianConfig (and other structures like it) to log during debug - so we need some way to implement `GuradianConfig.stringify()`. 

Each dataType needs to create its own part, and all together need to create a string that we can log (best if we can log it nicely both in production and in development with multi-line) - we are actually building a structure somewhat like a JSON  - so maybe we simply need to build a JSON. 

The depth used today is because we are building it to print more as a JSON prettify would do. Maybe better if we build a string that is a legal JSON and use some lib to prettify it when needed.

Reply:
I'd implement [the `json.Marshaller` interface](https://pkg.go.dev/encoding/json#Marshaler)

Reply:
Note that you also may want to implement `fmt.Stringer`, even if it just calls `MarshalJSON` under the covers.